lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -217,6 +231,85 @@ defmodule ExUnit.Callbacks do
     end
   end
 
+  @supervisor_opts [strategy: :one_for_one, max_restarts: 1_000_000, max_seconds: 1]
+
+  @doc """
+  Starts a child process under the test supervisor.
+
+  It expects a child specification or a module, similar to the ones
+  given to `Supervisor.start_link/2`. For example, if your application
+  starts a supervision tree by running:
+
+      Supervisor.start_link([MyServer, {OtherSupervisor, ...}], ...)
+
+  You can start those processes under test in isolation by running:
+
+      start_supervised(MyServer)
+      start_supervised({MyServer, :initial_value})
+
+  A keyword list can also be given if there is a need to change
+  the child specification for the given child process:
+
+      start_supervised({MyServer, :initial_value}, restart: :temporary)
+
+  See the `Supervisor` module for a discussion on child specifications
+  and the available specification keys.
+
+  The advantage of starting a process under the test supervisor is that
+  it is guaranteed to exit before the next test starts. Furthermore,
+  because the child process is supervised, it will be restarted in case
+  of crashes according to the `:restart` strategy in the child
+  specification, even if stopped manually. Therefore, to guarantee a
+  process start with `start_supervised/2` correctly terminates without
+  restarts, see `stop_supervised/1` instead.
+
+  This function returns `{:ok, pid}` in case of success, otherwise it
+  returns `{:error, reason}`.
+  """
+  @spec start_supervised(Supervisor.child_spec | module | {module, term}, keyword) ::
+        Supervisor.on_start_child
+  def start_supervised(child_spec_or_module, opts \\ []) do
+    sup =
+      case ExUnit.OnExitHandler.get_supervisor(self()) do
+        {:ok, nil} ->
+          {:ok, sup} = Supervisor.start_link([], @supervisor_opts)
+          ExUnit.OnExitHandler.put_supervisor(self(), sup)
+          sup
+        {:ok, sup} ->
+          sup
+        :error ->
+          raise ArgumentError, "start_supervised/2 can only be invoked from the test process"
+      end
+
+    Supervisor.start_child(sup, Supervisor.child_spec(child_spec_or_module, opts))
+  end
+
+  @doc """
+  Stops a child process started via `start_supervised/2`.
+
+  This function expects the `id` in the child specification.
+  For example:
+
+      {:ok, _} = start_supervised(MyServer)
+      :ok = stop_supervised(MyServer)
+
+  It returns `:ok` if there is a supervised process with such
+  `id`, `{:error, :not_found}` otherwise.
+  """
+  @spec stop_supervised(id :: term()) :: :ok | {:error, :not_found}
+  def stop_supervised(id) do
+    case ExUnit.OnExitHandler.get_supervisor(self()) do
+      {:ok, nil} ->
+        {:error, :not_found}
+      {:ok, sup} ->
+        with :ok <- Supervisor.terminate_child(sup, id),
+             :ok <- Supervisor.delete_child(sup, id),
+             do: :ok
+      :error ->
+        raise ArgumentError, "stop_supervised/1 can only be invoked from the test process"
+    end
```

Comment:
You could want to ignore the result though and that would not be possible if raising an error.